### ISSUE TITLE
Script for 8-way movement not smooth in EasyRPG

### ISSUE THREAD

**Issue by M3Kat** [EasyRPG/Player#1346]
#### Name of the game:
Test Project

#### Player platform:
Windows 32 & 64-Bit

#### Attached files:
https://www.dropbox.com/s/vcchdiyqiom6eow/Project1.zip?dl=1

Log :
```
[2018-03-27 08:08:22] Debug: EasyRPG Player 0.5.3 (git~73@0220488, 2018-03-09) started
[2018-03-27 08:08:22] Debug: =========================================================
[2018-03-27 08:08:22] Debug: CLI: 
[2018-03-27 08:08:22] Debug: Opened audio at 44100 Hz (stereo), format: S16LSB
[2018-03-27 08:08:22] Debug: Cannot find: Title/Title1 (!)
[2018-03-27 08:08:22] Debug: Detected encoding: ibm-5348_P100-1997. Files not found. Trying next.
[2018-03-27 08:08:22] Debug: Cannot find: Title/Title1 (!)
[2018-03-27 08:08:22] Debug: Detected encoding: UTF-8. Files not found. Trying next.
[2018-03-27 08:08:22] Debug: Cannot find: Title/Title1 (!)
[2018-03-27 08:08:22] Debug: Detected encoding: ibm-5346_P100-1998. Files not found. Trying next.
[2018-03-27 08:08:22] Debug: Cannot find: Title/呩瑬攱 (!)
[2018-03-27 08:08:22] Debug: Detected encoding: UTF-16BE. Files not found. Trying next.
[2018-03-27 08:08:22] Debug: Cannot find: Title/楔汴ㅥ (!)
[2018-03-27 08:08:22] Debug: Detected encoding: UTF-16LE. Files not found. Trying next.
[2018-03-27 08:08:22] Debug: Bad encoding: ISO-8859-9. Trying next.
[2018-03-27 08:08:22] Debug: Cannot find: Title/Title1 (!)
[2018-03-27 08:08:22] Debug: Detected encoding: ibm-943_P15A-2003. Files not found. Trying next.
[2018-03-27 08:08:22] Debug: Cannot find: Title/Title1 (!)
[2018-03-27 08:08:22] Debug: Detected encoding: windows-936-2000. Files not found. Trying next.
[2018-03-27 08:08:22] Debug: Bad encoding: EUC-JP. Trying next.
[2018-03-27 08:08:22] Debug: Cannot find: Title/Title1 (!)
[2018-03-27 08:08:22] Debug: Detected encoding: windows-949-2000. Files not found. Trying next.
[2018-03-27 08:08:22] Debug: Cannot find: Title/Title1 (!)
[2018-03-27 08:08:22] Debug: Detected encoding: Big5. Files not found. Trying next.
[2018-03-27 08:08:22] Debug: Bad encoding: IBM424_ltr. Trying next.
[2018-03-27 08:08:22] Debug: Bad encoding: IBM424_rtl. Trying next.
[2018-03-27 08:08:22] Debug: Detected encoding: ibm-5348_P100-1997
[2018-03-27 08:08:22] Debug: Using . as Game and Save directory
[2018-03-27 08:08:22] Debug: Loading game Untitled
[2018-03-27 08:08:22] Debug: Using RPG2k3 (English release, v1.11) Interpreter
[2018-03-27 08:08:22] Debug: RPG2k >= v1.50 / RPG2k3 >= v1.05 detected
[2018-03-27 08:08:22] Debug: Engine configured as: 2k=0 2k3=1 2k3Legacy=0 MajorUpdated=1 Eng=1
[2018-03-27 08:08:22] Debug: Adding C:\Program Files (x86)\Steam\steamapps\common\RPG Maker 2003\RTP to RTP path
[2018-03-27 08:08:22] Debug: Adding C:\Program Files (x86)\rpg2003\RTP to RTP path
[2018-03-27 08:08:22] Debug: Actor 1: Removing invalid item 44 (of type 2) from equipment slot 2 (needs type 1)
[2018-03-27 08:08:23] Debug: libpng: iCCP: known incorrect sRGB profile
[2018-03-27 08:08:23] Debug: libpng: cHRM: inconsistent chromaticities
[2018-03-27 08:08:25] Debug: Starting new game
[2018-03-27 08:08:25] Debug: Loading Map Map0001.lmu
[2018-03-27 08:08:25] Debug: Tree: MAP0001
[2018-03-27 08:08:36] Debug: Actor 1: Removing invalid item 44 (of type 2) from equipment slot 2 (needs type 1)
```

#### Video showcasing issue:
[](url)https://youtu.be/5m9MSr2wX10

#### Description:
This is an event allowing for 8-way movement. it works fine in the regular RM2k3, but as of march 26, 2018, the movement is not in the latest continuous version of EasyRPG. Maybe it has to do with the "Wait for All Movement" command.

**Comment by Ghabry** [comment]
Excellent bug report. Thanks.

This must be a problem with the "IsMovable" check because when I disable normal movement diagonal movement starts to work flawless.

**Comment by M3Kat** [comment]
Happy to know you found the issue!
I never thought about disabling the check since it inevitably leads to a softlock.

**Comment by Ghabry** [comment]
nope that was a red hering. The problem was that the script blocked for too long in "Wait for Movement".
A working fix appears to be removing "IsStopping" in:

https://github.com/EasyRPG/Player/blob/master/src/game_character.cpp#L432-L438

**Comment by Ghabry** [comment]
maybe fixed by #1601 . needs test (also -> 0.6.x)

**Comment by fmatthew5876** [comment]
This is fixed by #1601
